modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java

```diff
@@ -2571,9 +2571,7 @@ private GroupByStep _getOneToManyObjectEntriesGroupByStep(
 					}
 
 					return _getPermissionWherePredicate(
```

Comment:
Hey @p-albuquerque , do we have the same problem for many to many relationship?
I saw a that the same filter is used for this method `_getManyToManyObjectEntriesGroupByStep` and also for `getRelatedModelsPredicate` (which I think is used to search for a related entry)

Reply:
@guilhermedcamacho I verified that this situation also happens for manyToMany scenarios. I will append the fix and increase this scenario on test